extensions/amp-story/0.1/amp-story-auto-ads.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {user} from '../../../src/log';
+
+
+/** @const */
+// const EXPERIMENT = 'amp-story-auto-ad';
+
+/** @const */
+const TAG = 'amp-story-auto-ads';
+
+export class AmpStoryAutoAds extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+    this.ampStory = null;
+  }
+
+  /** @override */
+  buildCallback() {
+    const ampStory = this.element.parentElement;
+    user().assert(ampStory.tagName === 'AMP-STORY',
+        `<${TAG}> should be child of <amp-story>`);
+    this.ampStory = ampStory;
+  }
+
+  /** @override */
+  isLayoutSupported() {
+    return true;
+  }
+
+  /** @override */
+  layoutCallback() {
+    this.ampStory.appendChild(this.makeMockPage());
+  }
+
+  // temporary to be replaced with real page later
+  makeMockPage() {
+    const ampStoryAdPage = document.createElement('amp-story-page');
+    ampStoryAdPage.id = 'i-amphtl-ad-page-1';
```

Comment:
typo amphtml.

@newmuis what do you think about this?
`i-amphtml-` is reserved ID prefix by validator. we can have `i-amphtml-ad-` as a prefix for all auto generated pages for ads.

this ID will be consumed by analytics.

Reply:
LGTM.  My question is just whether there's something more valuable than a numerical ID that publishers may want for analytics purposes, but I don't understand enough about how publishers consume/evaluate ads to be able to answer that.

Reply:
that's a good point.

maybe we should include the previous pageId into the ID? If the previous page has an ID of `page-3`, the ad ID becomes: `i-amphtml-AD-page-3`

when reporting to analytics, we strip out the `i-amphtml-` prefix. so publisher will see `AD-page-3`, so he can infer where the ad is shown.

Of course, we will never show 2 ads next to each other, to make sure no ID conflicts between the ads.

Reply:
I like this idea, but I think to accomplish it we might need two pieces of identification, one to keep track of it before we know where it will be placed, and another that contains the placement data. Maybe we could use a different attr for one of these?